src/components/home/Contact.js

```diff
@@ -1,39 +1,153 @@
 import React, { Component } from 'react';
 import HorizSpinner from '../widgets/spinners/horiz-spinner'
 import $ from 'jquery'
+import axios from 'axios';
 
 class Contact extends Component {
 	
   constructor(props) {
     super(props);
     this.setSectionRef = this.setSectionRef.bind(this);
+    this.setSchoolFieldRef = this.setSchoolFieldRef.bind(this);
+    this.setEmailFieldRef = this.setEmailFieldRef.bind(this);
+    this.setSubmitMobileContainerRef = this.setSubmitMobileContainerRef.bind(this);
+    this.setDesktopBtnRef = this.setDesktopBtnRef.bind(this);
+    this.setMobileBtnRef = this.setMobileBtnRef.bind(this);
     this.getTopPosition = this.getTopPosition.bind(this);
+    this.onDesktopSubmit = this.onDesktopSubmit.bind(this);
+    this.onMobileSubmit = this.onMobileSubmit.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.parseFields = this.parseFields.bind(this);
+    this.isValidInfo = this.isValidInfo.bind(this);
+    this.sendContactInfo = this.sendContactInfo.bind(this);
+    this.onMobileComplete = this.onMobileComplete.bind(this);
+    this.onDesktopComplete = this.onDesktopComplete.bind(this);
+    this.clearContactFields = this.clearContactFields.bind(this);
+    this.removeInvalid = this.removeInvalid.bind(this);
   }
 
   setSectionRef(ref) {
     this.section = ref;
   }
   
+  setSchoolFieldRef(ref) {
+    this.schoolField = ref;
+  }
+  
+  setEmailFieldRef(ref) {
+    this.emailField = ref;
+  }
+  
+  setSubmitMobileContainerRef(ref) {
+    this.submitMobileContainer = ref;
+  }
+  
+  setDesktopBtnRef(ref) {
+    this.desktopBtn = ref;
+  }
+  
+  setMobileBtnRef(ref) {
+    this.mobileBtn = ref;
+  }
+  
 	getTopPosition() {
     return $(this.section)[0].offsetTop;
 	}
 	
+	onDesktopSubmit() {
+    this.onSubmit(false);
+  }
+  
+  onMobileSubmit() {
+    this.onSubmit(true);
+  }
+  
+  onSubmit(isMobile) {
+    var info = this.parseFields();
+    
+    if (this.isValidInfo(info, isMobile)) {
+      this.sendContactInfo(info, isMobile);
+    }
+  }
+  
+  parseFields() {
+    return {
+      school: $(this.schoolField).val().trim(),
+      email: $(this.emailField).val().trim()
+    };
+  }
+  
+  isValidInfo(info, isMobile) {
+    var isValid = true;
+  
+    for (var k in info) {
+      if (!info[k]) {
+        $('.contact-field[name=' + k + ']').addClass(isMobile ? 'invalid-mobile' : 'invalid');
+        isValid = false
+      }
+    }
+  
+    return isValid;
+  }
+  
+  sendContactInfo(info, isMobile) {
+    if (isMobile) {
+      $(this.submitMobileContainer).addClass('loading');
+    }
+    
+    isMobile ? this.onMobileComplete() : this.onDesktopComplete();
+    // axios.post('/api/contact', info).then((resp) => {
```

Comment:
This code would be enabled once we have an actual API right? Maybe explain that with a comment here.